libs/server-sdk/src/config/config_builder.cpp

```diff
@@ -39,14 +44,23 @@ tl::expected<Config, Error> ConfigBuilder::Build() const {
     if (!endpoints_config) {
         return tl::make_unexpected(endpoints_config.error());
     }
```

Comment:
In this scheme, users can set `Offline` as a top-level config property, which is the same as disabling events and the data system. If they need more granularity, then can individually enable/disable just events or the data system.